wagtail/admin/templates/wagtailadmin/generic/form.html

```diff
@@ -6,12 +6,6 @@
     <form action="{{ action_url }}" method="POST" novalidate{% if form.is_multipart %} enctype="multipart/form-data"{% endif %}>
         {% csrf_token %}
 
-        {% block non_field_errors %}
-            {% for error in form.non_field_errors %}
-                {% help_block status="critical" %}{{ error }}{% endhelp_block %}
-            {% endfor %}
-        {% endblock %}
-
```

Comment:
I think we need this, this removal will likely break all other forms.

Reply:
No, this was introduced in the PR that added case-insensitive duplicate detection to sites. I wrote an explanation in the issue.

Reply:
Ok. I'll have another read. I'm probably confusing the sign in form issue where we never showed non-field errors.

Reply:
@lb- I believe the error messages are currently being displayed twice while editing a site due to the use of the message framework https://github.com/wagtail/wagtail/blob/ade573bce28b43bd62bab017631c0d91d367ad85/wagtail/sites/views.py#L37 with non-field errors. To resolve this we need to use only one thing.

@lb- and @laymonage,  please let me know if any changes are needed in current PR.